### PR TITLE
Update HGG subtyping (4/11)

### PR DESCRIPTION
**Pull request by komalsrathi**
Created this branch off of [update-nf1-score](https://github.com/d3b-center/hope-cohort-analysis/tree/update-nf1-score) branch.

Comparison of base and annotated histologies file:

```
base_file <- read_tsv("data/Hope-GBM-histologies-base.tsv")
annotated_file <- read_tsv("analyses/molecular-subtyping-HGG/results/Hope-GBM-histologies.tsv")

# new samples
annotated_file %>%
  filter(sample_id %in% c("7316-3000", "7316-1106")) %>%
  dplyr::select(sample_id, Kids_First_Biospecimen_ID, experimental_strategy, cancer_group, cancer_group_short, molecular_subtype)

  sample_id Kids_First_Biospecimen_ID experimental_strategy cancer_group                  cancer_group_short molecular_subtype
  <chr>     <chr>                     <chr>                 <chr>                         <chr>              <chr>            
1 7316-1106 BS_96QPQX80               Methylation           Pleomorphic xanthoastrocytoma NA                 PXA              
2 7316-1106 BS_9XHV04WE               WGS                   NA                            NA                 NA               
3 7316-1106 BS_H1XPVS9A               RNA-Seq               Pleomorphic xanthoastrocytoma NA                 PXA              
4 7316-1106 BS_J4E9SW51               WGS                   Pleomorphic xanthoastrocytoma NA                 PXA              
5 7316-3000 BS_0TCRV9AC               WGS                   NA                            NA                 NA               
6 7316-3000 BS_ABZ3BK38               WGS                   High-grade glioma             HGG                HGG, H3 wildtype 
7 7316-3000 BS_JRYJGYC4               RNA-Seq               High-grade glioma             HGG                HGG, H3 wildtype 
8 7316-3000 BS_Z0KPQPN2               Methylation           NA                            NA                 NA                  
```

Wanted to ask what needs to be done for the PXAs?

### CODE REVIEW

**Comment by jharenza** [comment]
>Wanted to ask what needs to be done for the PXAs?

can you make these cancer_group_short == HGG?

**Comment by komalsrathi** [comment]
Added the condition `cancer_group_short is HGG where molecular_subtype is PXA`.

File: analyses/molecular-subtyping-HGG/03-molecular-subtype-integrate.R [hunk 1]
```diff
@@ -52,7 +52,13 @@ hist_with_subtype <- hist %>%
...
-  select(-short_histology) %>%
-  write_tsv(file.path(results_dir, "Hope-GBM-histologies.tsv"))
+  select(-short_histology) 
+
+# cancer_group_short is HGG where molecular_subtype is PXA
+hist_with_subtype %>%
+  mutate(cancer_group_short = ifelse(!is.na(molecular_subtype) & molecular_subtype == "PXA", "HGG", cancer_group_short)) 
+
+# write to file
+hist_with_subtype %>% write_tsv(file.path(results_dir, "Hope-GBM-histologies.tsv"))
```
Comment (naqvia) [inline analyses/molecular-subtyping-HGG/03-molecular-subtype-integrate.R:62]: I don't think you are saving these with the condition. I would suggest: 
```
hist_with_subtype %>%
  mutate(cancer_group_short = ifelse(!is.na(molecular_subtype) & molecular_subtype == "PXA", "HGG", cancer_group_short)) %>%
  write_tsv(file.path(results_dir, "Hope-GBM-histologies.tsv"))
```

**Review by naqvia — CHANGES_REQUESTED** [review]
Made a suggestion. Seemed like the intended file was not being saved properly.

File: analyses/molecular-subtyping-HGG/03-molecular-subtype-integrate.R (hunk 1 above)
Comment (komalsrathi) [reply analyses/molecular-subtyping-HGG/03-molecular-subtype-integrate.R:62]: Yes, agreed - missed the assignment. I will fix and rerun. Thanks for checking.

**Comment by komalsrathi** [comment]
FYI - This is the warning I was getting and seems to be only on docker. This was causing file reading to be interrupted way before and the output was incorrect. 

```
Avoidable 14.461 seconds. This file is very unusual: it ends abruptly without a final newline, and also its size is a multiple of 4096 bytes. Please properly end the last row with a newline using for example 'echo >> file' to avoid this  time to copy.
```

This was fixed by changing the `tmpdir` parameter to scratch dir (default was `/tmp`).

Also, I checked even with the same version of data.table and R, I don't get this on my local machine.

**Review by jharenza — COMMENTED** [review]
one comment about the dockerfile

File: analyses/molecular-subtyping-HGG/results/HGG_cleaned_mutation.tsv [hunk 2]
```diff
@@ -76,7 +76,7 @@ PT_RSDTCFDA	7316-204	BS_SK4H5MJQ	No	No	No	No	No	No	No	No	No	No	None	No R132 or R
 PT_S2BEPYQ0	7316-3000	BS_ABZ3BK38	No	No	No	No	No	No	No	No	No	No	None	No R132 or R172	None	No V600E
 PT_SDPA6AAP	7316-624	BS_SMHNXC5G	No	No	No	No	No	No	No	No	No	No	None	No R132 or R172	None	No V600E
 PT_SYHB12RN	7316-194	BS_9DM8H1RX	No	No	No	No	No	No	No	No	No	No	FGFR1	No R132 or R172	None	No V600E
-PT_T8V9ES93	7316-1106	BS_J4E9SW51	No	No	No	No	No	No	No	No	No	No	None	No R132 or R172	None	No V600E
+PT_T8V9ES93	7316-1106	BS_J4E9SW51	No	No	No	No	No	No	No	No	No	No	BRAF	No R132 or R172	None	p.V600E
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/results/HGG_cleaned_mutation.tsv:79]: noting this is expected.

File: analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv [hunk 3]
```diff
@@ -183,11 +183,11 @@ PT_C7HHJG7Z	BS_9ZJT0CP0	7316-4516	DMG, H3 K28, TP53	DMG, H3 K28	loss
...
-PT_CB1DN0V8	BS_3J4X955P	7316-2241	HGG, H3 wildtype, TP53	NA	loss
-PT_CB1DN0V8	BS_84XCR9GG	7316-2241	HGG, H3 wildtype, TP53	NA	loss
-PT_CB1DN0V8	BS_N5VEQ8RH	7316-2241	HGG, H3 wildtype, TP53	NA	loss
-PT_CB1DN0V8	BS_TNVSRPSV_PHOS	7316-2241	HGG, H3 wildtype, TP53	NA	loss
-PT_CB1DN0V8	BS_TNVSRPSV_WCP	7316-2241	HGG, H3 wildtype, TP53	NA	loss
+PT_CB1DN0V8	BS_3J4X955P	7316-2241	HGG, H3 wildtype	NA	other
+PT_CB1DN0V8	BS_84XCR9GG	7316-2241	HGG, H3 wildtype	NA	other
+PT_CB1DN0V8	BS_N5VEQ8RH	7316-2241	HGG, H3 wildtype	NA	other
+PT_CB1DN0V8	BS_TNVSRPSV_PHOS	7316-2241	HGG, H3 wildtype	NA	other
+PT_CB1DN0V8	BS_TNVSRPSV_WCP	7316-2241	HGG, H3 wildtype	NA	other
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv:190]: I think once this is rerun with changes from TP53 we should see TP53 come back here

File: analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv [hunk 4]
```diff
@@ -196,78 +196,78 @@ PT_CSZ9QA1N	BS_R3308SP5	7316-371	HGG, H3 wildtype, TP53	NA	activated
...
-PT_JNEV57VK	BS_VF5XWFXD	7316-2594	DMG, H3 K28, TP53	NA	loss
+PT_JC4DHFJR	BS_75H9VE5T	7316-2536	DMG, H3 K28	DMG, H3 K28	other
+PT_JC4DHFJR	BS_7JVKJPB7	7316-2536	DMG, H3 K28	DMG, H3 K28	other
+PT_JC4DHFJR	BS_QB84TBA3	7316-2536	DMG, H3 K28	DMG, H3 K28	other
+PT_JC4DHFJR	BS_R0KFZRQN_PHOS	7316-2536	DMG, H3 K28	DMG, H3 K28	other
+PT_JC4DHFJR	BS_R0KFZRQN_WCP	7316-2536	DMG, H3 K28	DMG, H3 K28	other
+PT_JNEV57VK	BS_85Q5P8GF	7316-2594	DMG, H3 K28	NA	other
+PT_JNEV57VK	BS_AMDGBEMH_PHOS	7316-2594	DMG, H3 K28	NA	other
+PT_JNEV57VK	BS_AMDGBEMH_WCP	7316-2594	DMG, H3 K28	NA	other
+PT_JNEV57VK	BS_VF5XWFXD	7316-2594	DMG, H3 K28	NA	other
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv:258]: similarly here

File: analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv [hunk 5]
```diff
@@ -375,81 +373,81 @@ PT_QQZXYR4M	BS_10MCSWE4_WCP	7316-1946	HGG, IDH, TP53	NA	loss
...
+PT_SDPA6AAP	BS_845CYEJC_WCP	7316-624	HGG, H3 wildtype	NA	other
+PT_SDPA6AAP	BS_JJZKPXCK	7316-624	HGG, H3 wildtype	NA	other
+PT_SDPA6AAP	BS_SMHNXC5G	7316-624	HGG, H3 wildtype	NA	other
+PT_SYHB12RN	BS_9DM8H1RX	7316-194	HGG, H3 wildtype	NA	other
+PT_SYHB12RN	BS_AAVNFHHB	7316-194	HGG, H3 wildtype	NA	other
+PT_SYHB12RN	Prot_7316-194_PHOS	7316-194	HGG, H3 wildtype	NA	other
+PT_SYHB12RN	Prot_7316-194_WCP	7316-194	HGG, H3 wildtype	NA	other
+PT_T8V9ES93	BS_96QPQX80	7316-1106	PXA	NA	other
+PT_T8V9ES93	BS_H1XPVS9A	7316-1106	PXA	NA	other
+PT_T8V9ES93	BS_J4E9SW51	7316-1106	PXA	NA	other
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/results/Hope_subtype.tsv:404]: correct

File: doc/release-notes.md [hunk 6]
```diff
@@ -1,7 +1,21 @@
 # Release note
 
+## Current release (V3)
+- release date: 2024-08-16
+- Overview:
+  - Sequence sample:
+    - 87 RNA-seq
+    - 157 WGS
+    - 80 DNA methylation
```
Comment (jharenza) [inline doc/release-notes.md:9]: need to merge master and these will also go away

**Pull request merged**